packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -3715,6 +3715,9 @@ namespace Tpetra {
       }
       throw std::runtime_error(msg);
     }
+#if defined(KOKKOS_ENABLE_CUDA)
+    Kokkos::fence();
+#endif
     owningView_.sync_host();
```

Comment:
@kyungjoo-kim - why ifdef for CUDA here? This seems like this will make porting to new node types similar to cuda painful to debug. I would prefer to see a fence here without the ifdefs or let's figure out if there is a fix on the nox side. Is nox doing something unexpected or unsupported?

Reply:
@rppawlo This is just testing and temporary. We still do not exactly know what causes the tests failing. With refactoring the multivector, we remove explicit sync/modify interface from users and downstream packages. The getLocalView interface takes care of the multi memory interface. Without UVM, the dual view sync gives synchronization point (I do not test myself for other packages but tpetra tests pass with this refactoring). With UVM, the dual view syntax should sync the device although it does not perform data transfer. In this case, the above fence won't be necessary. However, some tests in nox, teko and muelu as specified in the previous PR tests fail without the above explicit fence. This is a kind of double fencing as the next dual view sync should put a fence when it is UVM. We need to investigate more to understand the dual view logic better.

Reply:
are you guys actually using multiple streams anywhere? Just to make sure. I think I convinced myself that UVM DualView sync should fence globally. But it should only make a difference if you actually were using multiple streams

Reply:
Just to verify: this is the case when the DualView is UVM? It looks like it happens any time CUDA is snabled, which sounds too broad

Reply:
@crtrott I thought that UVM itself uses multiple streams.

Reply:
@crtrott BTW, I use multiple streams in the ifpack2 block tridiagonal solver. Ifpack2 tests are passed and I am not sure if it is related to the previous PR test failures on the other packages.